AutomatedTesting/Gem/PythonTests/scripting/scripting_utils/script_canvas_component.py

```diff
@@ -65,6 +61,22 @@ def create_new_entity_with_component(self, entity_name: str, sc_file_path: str,
         self.script_canvas_component = self.editor_entity.get_components_of_type([SCRIPT_CANVAS_UI])[0]
         self.script_canvas_component.set_component_property_value(SCRIPT_CANVAS_COMPONENT_PROPERTY_PATH, sourcehandle)
 
+    def change_component_graph_file(self, sc_file_path: str):
+        """
+        Function for updating the script canvas file being used by the component
+
+        param sc_file_path : the string path the new graph file on disk
+
+        returns none
+        """
+        sourcehandle = scriptcanvas.SourceHandleFromPath(sc_file_path)
+        old_value = self.script_canvas_component.get_component_property_value(SCRIPT_CANVAS_COMPONENT_PROPERTY_PATH)
+        self.script_canvas_component.set_component_property_value(SCRIPT_CANVAS_COMPONENT_PROPERTY_PATH, sourcehandle)
+
+        set_value = self.script_canvas_component.get_component_property_value(SCRIPT_CANVAS_COMPONENT_PROPERTY_PATH)
+        assert set_value != old_value and set_value is not None, f"Graph file could not be set! {sc_file_path} not found."
+
+
```

Comment:
Let's refactor this into using the model I've set in `editor component behaviors`

Reply:
updated the class to use the validation file